### ISSUE TITLE
Header file is not installed when SIMDJSON_JUST_LIBRARY=ON

### ISSUE THREAD

**Issue by myd7349** [simdjson/simdjson#1000]
Before submitting an issue, please ensure that you have read the documentation:

* Basics is an overview of how to use simdjson and its APIs: https://github.com/simdjson/simdjson/blob/master/doc/basics.md
* Performance shows some more advanced scenarios and how to tune for them: https://github.com/simdjson/simdjson/blob/master/doc/performance.md
* Contributing: https://github.com/simdjson/simdjson/blob/master/CONTRIBUTING.md

**Describe the bug**
When `SIMDJSON_JUST_LIBRARY=ON` is specified, this line will be skiped:
https://github.com/simdjson/simdjson/blob/6b85b9a416ca464289bb5673ce277b5bbf3f912a/CMakeLists.txt#L46

as a result, the header file of simdjson won't be installed.

The log:
https://dev.azure.com/vcpkg/public/_build/results?buildId=39425&view=logs&j=7b75bd19-17d3-53d4-00fd-23f1a49a8ba4&t=a23dba4e-3a62-59dc-a73c-a7222676d7a4
```
-- Configuring x64-osx-rel
-- Building x64-osx-dbg
-- Building x64-osx-rel
-- Installing: /Users/vagrant/Data/packages/simdjson_x64-osx/share/simdjson/copyright
-- Performing post-build validation
The folder /include is empty or not present. This indicates the library was not correctly installed.
Found 1 error(s). Please correct the portfile:
    /Volumes/data/work/1/s/ports/simdjson/portfile.cmake
-- Performing post-build validation done
Error: Building package simdjson:x64-osx failed with: POST_BUILD_CHECKS_FAILED
Elapsed time for package simdjson:x64-osx: 14.05 s

```

**To Reproduce**
```
mkdir build
pushd build
cmake .. -DSIMDJSON_JUST_LIBRARY=ON -DCMAKE_INSTALL_PREFIX=./install
make install
```

**Configuration (please complete the following information if relevant):**
 - OS: [e.g. Ubuntu 16.04.6 LTS]
 - Compiler [e.g. Apple clang version 11.0.3 (clang-1103.0.32.59) x86_64-apple-darwin19.4.0]

**Indicate whether you are willing or able to provide a bug fix as a pull request**

If you plan to contribute to simdjson, please read our 
* CONTRIBUTING guide: https://github.com/simdjson/simdjson/blob/master/CONTRIBUTING.md and our
* HACKING guide: https://github.com/simdjson/simdjson/blob/master/HACKING.md

**Comment by lemire** [comment]
Verified. We had a test, but the test was broken apparently.

I will issue a new patch release right now (0.4.2).

**Comment by lemire** [comment]
(It is a bug.)

**Comment by lemire** [comment]
Fixed in version 0.4.2.
